### ISSUE TITLE
Feature request > Fix `quoted-strings`

### ISSUE THREAD

**Issue by giggio** [opt-nc/yamlfixer#195]
# ❔ Context

Fix `quoted-strings`. Docs:
https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.quoted_strings

# 💡 Proposed solution

1. Change `quote-type` to double or single accordingly;
1. Remove unnecessary quotes when `required` is set to `only-when-needed`;
1. Add quotes when `required` is set to true;
1. Remove quoted quotes when `allow-quoted-quotes` is false.

**Comment by adriens** [comment]
Hi @tamere-allo-peter , what's your opinion about this feature request ?

**Comment by tamere-allo-peter** [comment]
@adriens very complex. Currently we don't use yamllint's configuration file, we only pass it as-is to yamllint when needed, and we "fix" by doing what yamllint suggests, whenever it does suggest something, and do our best the remaining of the time. Closing this one is very hard work because if we begin to parse yamllint's configuration file, this opens the door to a number of other fixes we should do according to this file that we don't right now.

**Comment by adriens** [comment]
So should we flag this as a `wontfix` because of lack of time ?

**Comment by tamere-allo-peter** [comment]
@adriens we'll leave it open for now, maybe I'll have more time for this at the end of the year.

**Comment by adriens** [comment]
So it will stay on : 
![image](https://user-images.githubusercontent.com/5235127/194218111-b0a0e10f-24bd-4ea0-acf0-d5249df1c0e4.png)

⏸️

**Comment by fortin-alex** [comment]
> this opens the door to a number of other fixes we should do according to this file that we don't right now.

Perhaps that it opens the door to different fixes, but the scope of the current github issue would remain "to automatically fix `quoted-strings`" and so the other fixes that could be made can be put in a backlog for future contributions.

This said, another issue would need to be created & resolved first (i.e. `Adding a yamllint configuration parser`)

ps: thanks for this package, it's really great!

**Comment by fortin-alex** [comment]
Re-reading my previous comment, it might be something that `yamllint` can fix on their end (i.e. they could write a better message when reporting a `quoted-strings` error)

On my end, I see this message from `yamllint`:
```
>>> cat .yamllint
rules:
 quoted-strings:
    quote-type: double
    required: true
    allow-quoted-quotes: true

>>> yamllint --format parsable .
./path/to/file/config.yaml:31:11: [error] string value is not quoted with double quotes (quoted-strings)
```

And I am wondering if this message is informative enough to be actionable? it seems to suggest to replace the single quotes by double quotes or to add double-quotes.

**Comment by tamere-allo-peter** [comment]
Yes it seems to be informative enough in the general case, however unfortunately I don't have the time to work on this project at the moment.
